### PR TITLE
[ALLUXIO-1322][WIP] Prepared classes for MissingCtor rule in Checkstyle

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-1322

Prepared some classes for the upcoming MissingCtor rule in Checkstyle. It checks that classes (except abstract ones) define a constructor and don't rely on the default one. The rule will look like the following

<module name="MissingCtor"/>

This ticket is WIP and needs some additional PRs.

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/lineage/LineageMasterClientRestServiceHandler.java [hunk 1]
```diff
@@ -57,6 +57,9 @@
 
   private final LineageMaster mLineageMaster = AlluxioMaster.get().getLineageMaster();
 
+  // prevent instantiation
+  private LineageMasterClientRestServiceHandler() {}
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/lineage/LineageMasterClientRestServiceHandler.java]: nit: `private LineageMasterClientRestServiceHandler() {} // prevent instantiation`

File: examples/src/main/java/alluxio/examples/keyvalue/KeyValueStoreQuickStart.java [hunk 2]
```diff
@@ -21,6 +21,10 @@
  * A quick start tutorial for creating a key-value store, putting and getting some key-value pairs.
  */
 public final class KeyValueStoreQuickStart {
+
+  // prevent instantiation
```
Comment (jsimsa) [inline examples/src/main/java/alluxio/examples/keyvalue/KeyValueStoreQuickStart.java]: ditto

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionReader.java [hunk 3]
```diff
@@ -31,6 +31,10 @@
    * Factory for {@link KeyValuePartitionReader}.
    */
   class Factory {
+
+    // prevent instantiation
```
Comment (jsimsa) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionReader.java]: ditto

File: keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java [hunk 4]
```diff
@@ -34,6 +34,10 @@
    * Factory for {@link KeyValuePartitionWriter}.
    */
   class Factory {
+
+    // prevent instantiation
```
Comment (jsimsa) [inline keyvalue/client/src/main/java/alluxio/client/keyvalue/KeyValuePartitionWriter.java]: ditto

**Comment by jsimsa** [comment]
@HorizonNet thanks, left some minor comments

**Comment by HorizonNet** [comment]
@jsimsa Thanks for the comments. I updated the PR accordingly.

**Comment by jsimsa** [comment]
LGTM

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**